cpp/ql/src/semmle/code/cpp/dataflow/internal/DataFlowImpl.qll

```diff
@@ -114,6 +114,32 @@ abstract class Configuration extends string {
    */
   predicate hasFlowToExpr(DataFlowExpr sink) { hasFlowTo(exprNode(sink)) }
 
+  /**
+   * Gets the exploration limit for `hasPartialFlow` measured in approximate
+   * number of interprocedural steps.
+   */
+  int explorationLimit() { none() }
```

Comment:
Is this the right default? Since it's only used from `hasPartialFlow`, we're effectively forcing all users to provide an additional override at all times. How about limiting to, say, 30 (100?) by default, and changing the qldoc to suggest lowering the limit if it performs poorly?

Reply:
This is in fact exactly the right default.  If you're using this feature and you have more than one configuration in scope, then any other default means that you can get massive performance problems due to exploration happening in other unrelated configurations (I got bitten by this during my testing).